src/compiler/checker.ts

```diff
@@ -1481,7 +1484,21 @@ namespace ts {
                     result = lookup(globals, name, meaning);
                 }
             }
-
+            if (!result) {
+                if (originalLocation && isInJavaScriptFile(originalLocation) && originalLocation.parent) {
+                    if (isRequireCall(originalLocation.parent, /*checkArgumentIsStringLiteralLike*/ false)) {
+                        return requireSymbol;

```

Comment:
we shoudl change all other uses in the checker for `isRequireCall` to use the symbol identity check instead.